### ISSUE TITLE
CRAN check FAIL due to 'pragmas in C/C++ headers and code'

### ISSUE THREAD

**Issue by Enchufa2** [RcppCore/Rcpp#788]
[See the CRAN checks](https://cran.r-project.org/web/checks/check_results_Rcpp.html).

**Comment by coatless** [comment]
Results from CRAN page

<details>CRAN checks:

Version: 0.12.14 
Check: foreign function calls 
Result: NOTE 
    Registration problem:
     symbol ‘symbol’ in the local frame:
     .Call(symbol)
    See chapter ‘System and foreign language interfaces’ in the ‘Writing R
    Extensions’ manual. 
Flavors: r-devel-linux-x86_64-debian-clang, r-devel-linux-x86_64-debian-gcc, r-devel-linux-x86_64-fedora-clang, r-devel-linux-x86_64-fedora-gcc, r-patched-linux-x86_64, r-release-linux-x86_64

Version: 0.12.14 
Check: pragmas in C/C++ headers and code 
Result: FAIL 
     
Flavors: r-devel-linux-x86_64-debian-clang, r-devel-linux-x86_64-debian-gc</details>

**Comment by eddelbuettel** [comment]
Are you implying we don't run `R CMD check` or that CRAN might not have told us?   Is it about the two FAIL under r-devel?  I would first blame R-devel, then the compilers.  

I suggest you close this.  Why don't you ask on rcpp-devel next time if it is an informal question?

**Comment by Enchufa2** [comment]
There was a recent [policy change](https://github.com/eddelbuettel/crp/blob/master/txt/policies.r3776.txt#L223):

> Packages should not attempt to disable compiler diagnostics.

It seems that pragmas are not allowed anymore.

**Comment by eddelbuettel** [comment]
You are referring to my service. I follow it.   

And we don't set pragmas to disable compiler diagnostics. Tinyformat does, albeit within `#ifdef _MSC_VER` so ...   Worst case, if we had to we would remove it.  We are in talks to update tinyformat anyway.

I'll close this now as I have work to do.  Appreciate the willingness to help, but again: maybe rcpp-devel.

**Comment by Enchufa2** [comment]
Ok, I'll begin with rcpp-devel next time. 

Let me add just one more comment, since the issue is already here: all packages (with or without pragmas) that depend on Rcpp are failing on those flavors.

**Comment by eddelbuettel** [comment]
You mean r-devel?  Talk to CRAN / R Core / r-package-devel. Not my beef.

**Comment by eddelbuettel** [comment]
I don't want to sound dismissive but they _are_ making huge changes now.  Check out what I stuck into the digest package -- ability to change the serialization.  Due to Luke and Tomas asking because they are about to make changes.  

So I think assuming that _r-devel_ may be stable right now is probably not a great starting point.

**Comment by Enchufa2** [comment]
I perfectly understand your point, no problem. It's just I saw [this commit](https://github.com/wch/r-source/commit/b76c8fd355a0f5b23d42aaf44a879cac0fc31fa4) from 3 days ago and today I started to receive emails about FAILs on CRAN, and I thought I should notify this. Because if CRAN implements this check as a requirement to submit a package, more than a thousand packages will be unable to update.

But you are right: I'll open a thread on r-package-devel right away.
